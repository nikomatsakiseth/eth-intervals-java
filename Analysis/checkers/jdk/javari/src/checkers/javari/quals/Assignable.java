package checkers.javari.quals;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import checkers.javari.JavariChecker;

import static java.lang.annotation.ElementType.*;

/**
 * Indicates that a field is assignable, even if it is inside a {@link ReadOnly}
 * instance.
 *
 * <p>
 *
 * This annotation is part of the Javari language.
 *
 * @see JavariChecker
 * @checker.framework.manual #javari-checker Javari Checker
 */
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({FIELD})
public @interface Assignable {

}
