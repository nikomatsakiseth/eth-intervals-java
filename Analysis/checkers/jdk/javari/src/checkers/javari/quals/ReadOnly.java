package checkers.javari.quals;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import checkers.javari.JavariChecker;
import checkers.quals.SubtypeOf;
import checkers.quals.TypeQualifier;
import static java.lang.annotation.ElementType.*;

/**
 * Indicates that, for the variable on which this annotation appears,
 * the object to which this variable refers will not be modified via
 * this reference, except its fields explicitly marked as
 * {@link Mutable}.
 *
 * <p>
 *
 * This annotation is part of the Javari language.
 *
 * @see Mutable
 * @see JavariChecker
 * @checker.framework.manual #javari-checker Javari Checker
 */
@Documented
@Retention(RetentionPolicy.RUNTIME)
//@Target({FIELD, LOCAL_VARIABLE, METHOD, PARAMETER, TYPE})
@TypeQualifier
@SubtypeOf({})
public @interface ReadOnly {

}
