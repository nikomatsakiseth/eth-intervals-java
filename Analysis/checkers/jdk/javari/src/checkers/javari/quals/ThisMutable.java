package checkers.javari.quals;

import java.lang.annotation.Target;

import checkers.quals.SubtypeOf;
import checkers.quals.TypeQualifier;

/**
 * An annotation used to represent a place holder immutability type, that is
 * equivalent to the ThisMutable type on the Javari typesystem.
 *
 * However, it is an implementation detail; hence, the package-scope.
 */
@TypeQualifier
@Target({}) // A programmer cannot write @ThisMutable in a program
@SubtypeOf(ReadOnly.class)
public @interface ThisMutable {

}
