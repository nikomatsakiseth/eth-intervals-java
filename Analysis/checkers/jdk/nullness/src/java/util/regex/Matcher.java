package java.util.regex;

import checkers.nullness.quals.*;

public final class Matcher implements java.util.regex.MatchResult {
  protected Matcher() {}
  public @NonNull java.util.regex.Pattern pattern() { throw new RuntimeException("skeleton method"); }
  public @NonNull java.util.regex.MatchResult toMatchResult() { throw new RuntimeException("skeleton method"); }
  public @NonNull java.util.regex.Matcher usePattern(@NonNull java.util.regex.Pattern a1) { throw new RuntimeException("skeleton method"); }
  public @NonNull java.util.regex.Matcher reset() { throw new RuntimeException("skeleton method"); }
  public @NonNull java.util.regex.Matcher reset(java.lang.CharSequence a1) { throw new RuntimeException("skeleton method"); }
  public int start() { throw new RuntimeException("skeleton method"); }
  public int start(int a1) { throw new RuntimeException("skeleton method"); }
  public int end() { throw new RuntimeException("skeleton method"); }
  public int end(int a1) { throw new RuntimeException("skeleton method"); }
  public @NonNull java.lang.String group() { throw new RuntimeException("skeleton method"); }
  public @Nullable java.lang.String group(int a1) { throw new RuntimeException("skeleton method"); }
  public int groupCount() { throw new RuntimeException("skeleton method"); }
  public boolean matches() { throw new RuntimeException("skeleton method"); }
  public boolean find() { throw new RuntimeException("skeleton method"); }
  public boolean find(int a1) { throw new RuntimeException("skeleton method"); }
  public boolean lookingAt() { throw new RuntimeException("skeleton method"); }
  public static @NonNull java.lang.String quoteReplacement(@NonNull java.lang.String a1) { throw new RuntimeException("skeleton method"); }
  public @NonNull java.util.regex.Matcher appendReplacement(@NonNull java.lang.StringBuffer a1, @NonNull java.lang.String a2) { throw new RuntimeException("skeleton method"); }
  public @NonNull java.lang.StringBuffer appendTail(@NonNull java.lang.StringBuffer a1) { throw new RuntimeException("skeleton method"); }
  public @NonNull java.lang.String replaceAll(@NonNull java.lang.String a1) { throw new RuntimeException("skeleton method"); }
  public @NonNull java.lang.String replaceFirst(@NonNull java.lang.String a1) { throw new RuntimeException("skeleton method"); }
  public @NonNull java.util.regex.Matcher region(int a1, int a2) { throw new RuntimeException("skeleton method"); }
  public int regionStart() { throw new RuntimeException("skeleton method"); }
  public int regionEnd() { throw new RuntimeException("skeleton method"); }
  public boolean hasTransparentBounds() { throw new RuntimeException("skeleton method"); }
  public @NonNull java.util.regex.Matcher useTransparentBounds(boolean a1) { throw new RuntimeException("skeleton method"); }
  public boolean hasAnchoringBounds() { throw new RuntimeException("skeleton method"); }
  public @NonNull java.util.regex.Matcher useAnchoringBounds(boolean a1) { throw new RuntimeException("skeleton method"); }
  public @NonNull java.lang.String toString() { throw new RuntimeException("skeleton method"); }
  public boolean hitEnd() { throw new RuntimeException("skeleton method"); }
  public boolean requireEnd() { throw new RuntimeException("skeleton method"); }
}
