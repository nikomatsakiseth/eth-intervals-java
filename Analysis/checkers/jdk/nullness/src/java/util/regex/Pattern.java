package java.util.regex;

import checkers.nullness.quals.*;

public final class Pattern implements java.io.Serializable{
    private static final long serialVersionUID = 0L;
  protected Pattern() {}
  public final static int UNIX_LINES = 1;
  public final static int CASE_INSENSITIVE = 2;
  public final static int COMMENTS = 4;
  public final static int MULTILINE = 8;
  public final static int LITERAL = 16;
  public final static int DOTALL = 32;
  public final static int UNICODE_CASE = 64;
  public final static int CANON_EQ = 128;
  public static @NonNull java.util.regex.Pattern compile(java.lang.String a1) { throw new RuntimeException("skeleton method"); }
  public static @NonNull java.util.regex.Pattern compile(java.lang.String a1, int a2) { throw new RuntimeException("skeleton method"); }
  public java.lang.String pattern() { throw new RuntimeException("skeleton method"); }
  public java.lang.String toString() { throw new RuntimeException("skeleton method"); }
  public @NonNull java.util.regex.Matcher matcher(java.lang.CharSequence a1) { throw new RuntimeException("skeleton method"); }
  public int flags() { throw new RuntimeException("skeleton method"); }
  public static boolean matches(java.lang.String a1, java.lang.CharSequence a2) { throw new RuntimeException("skeleton method"); }
  public java.lang.String[] split(java.lang.CharSequence a1, int a2) { throw new RuntimeException("skeleton method"); }
  public java.lang.String[] split(java.lang.CharSequence a1) { throw new RuntimeException("skeleton method"); }
  public static java.lang.String quote(java.lang.String a1) { throw new RuntimeException("skeleton method"); }
}
