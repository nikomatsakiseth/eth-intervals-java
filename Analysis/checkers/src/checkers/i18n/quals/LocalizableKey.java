package checkers.i18n.quals;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import checkers.quals.SubtypeOf;
import checkers.quals.TypeQualifier;
import checkers.quals.Unqualified;

/**
 * Indicates that the {@code String} type has been localized and
 * formatted for the target output locale
 */
@TypeQualifier
@SubtypeOf(Unqualified.class)
@Documented
@Retention(RetentionPolicy.RUNTIME)
public @interface LocalizableKey { }
