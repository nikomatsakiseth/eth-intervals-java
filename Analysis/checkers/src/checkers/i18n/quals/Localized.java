package checkers.i18n.quals;

import static com.sun.source.tree.Tree.Kind.*;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import checkers.quals.*;

/**
 * Indicates that the {@code String} type has been localized and
 * formatted for the target output locale
 */
@TypeQualifier
@SubtypeOf(Unqualified.class)
@ImplicitFor( trees = {
        /* All integer literals */
        INT_LITERAL,
        LONG_LITERAL,
        FLOAT_LITERAL,
        DOUBLE_LITERAL,
        BOOLEAN_LITERAL,

        //CHAR_LITERAL,
        //STRING_LITERAL,
        //NULL_LITERAL
        }
)
@Documented
@Retention(RetentionPolicy.RUNTIME)
public @interface Localized { }
