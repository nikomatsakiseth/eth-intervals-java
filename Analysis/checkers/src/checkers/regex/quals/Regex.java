package checkers.regex.quals;

import java.lang.annotation.*;

import checkers.quals.*;

/**
 * Indicates that a {@code String} is a valid regular expression.
 */
@Documented
@TypeQualifier
@Inherited
@SubtypeOf({Unqualified.class})
@Retention(RetentionPolicy.RUNTIME)
public @interface Regex {}
