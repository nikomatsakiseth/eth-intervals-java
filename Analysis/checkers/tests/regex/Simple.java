import checkers.regex.quals.Regex;

import java.util.regex.Pattern;

public class Simple {

    void regString() {
        String s1 = "validRegex";
        String s2 = "(InvalidRegex";
    }

    void validRegString() {
        @Regex String s1 = "validRegex";
        @Regex String s2 = "(InvalidRegex";    // error
    }

    void compileCall() {
        Pattern.compile("test.*[^123]$");
        Pattern.compile("$test.*[^123");    // error
    }

    void requireValidReg(@Regex String reg, String nonReg) {
        Pattern.compile(reg);
        Pattern.compile(nonReg);    // error
    }

    void testAddition(@Regex String reg, String nonReg) {
        @Regex String s1 = reg;
        @Regex String s2 = reg + "d.*sf";
        @Regex String s3 = reg + reg;

        @Regex String n1 = nonReg;     // error
        @Regex String n2 = reg + "(df";    // error
        @Regex String n3 = reg + nonReg;   // error

        @Regex String o1 = nonReg;     // error
        @Regex String o2 = nonReg + "sdf";     // error
        @Regex String o3 = nonReg + reg;     // error

    }
}