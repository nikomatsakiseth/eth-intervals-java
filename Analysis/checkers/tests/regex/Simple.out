:14: (type.incompatible)
:19: (type.incompatible)
:24: (type.incompatible)
:32: (type.incompatible)
:33: (type.incompatible)
:34: (type.incompatible)
:36: (type.incompatible)
:37: (type.incompatible)
:38: (type.incompatible)
